clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -497,6 +497,45 @@ void processCmakeMinimumRequired(std::string &Input, size_t &Size,
   Index = Begin + ReplStr.size(); // Update index
 }
 
+void processExecuteProcess(std::string &Input, size_t &Size, size_t &Index) {
+  std::string VarName;
+  std::string Value;
+  size_t Begin, End;
+
+  // Get the begin of firt argument
+  Index = skipWhiteSpaces(Input, Index);
+  Begin = Index;
+
+  Index = gotoEndOfCmakeCommandStmt(Input, Index);
+  End = Index;
+
+  // Get the value of the second argument
+  Value = Input.substr(Begin, End - Begin);
+
+  size_t Pos = Value.find("-Xcompiler");
+  if (Pos != std::string::npos) {
+
+    size_t NextPos = Pos + strlen("-Xcompiler");
+    NextPos = skipWhiteSpaces(Value, NextPos);
+
+    // To check if the value of opition "-Xcompiler" is a string literal, if it
+    // is a string literal, just remove '"' for the value.
```

Comment:
Is the usage like `-Xcompiler " --option_a --option_b \"balabala\" "` legal?
Do we need consider the nested ""?

Reply:
Currently, no nested option value is found in the real application for option"-Xcompiler", we currently consider scenario that the value is not nested.

Reply:
@zhimingwang36 The -Xcompiler option lets user to pass specific compiler options to the host compiler. For example:
```nvcc -Xcompiler "-Wall -O2" foo.cu  -c -o your_executable```, in cmake it use like this:
```
            execute_process(
                COMMAND ${NVCC_CMD} -Xcompiler "-dumpfullversion -dumpversion"
                OUTPUT_VARIABLE CUDA_CCVER
                ERROR_QUIET
            )
```
to get the version of the host compiler, it can accept formal like:
nvcc .c -ccbin gcc -Xcompiler "\"-dumpfullversion\" \"-dumpversion\""
nvcc .c -ccbin gcc -Xcompiler "-dumpfullversion -dumpversion"
Cannot accept:
nvcc .c -ccbin gcc -Xcompiler "\"-dumpfullversion -dumpversion\""

For our process logic we only remove the outermost layer“， the more than one options passed by \", like 
```nvcc .c -ccbin gcc -Xcompiler "\"-dumpfullversion -dumpversion\""```, the nvcc will emit error.